client/pages/chat/[id].jsx

```diff
@@ -1,19 +1,43 @@
+import cookie from "js-cookie";
 import { NextSeo } from "next-seo";
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 import client from "../../apollo-client";
 import ChatInfo from "../../components/ChatInfo";
 import { localesArr } from "../../content/locale";
 import { GET_GROUPCHAT, GET_GROUPCHAT_IDS } from "../../gql/GroupChat";
+import { GET_USER } from "../../gql/User";
+
+export default function Chat({ chat, id }) {
+  const [editPermissions, setEditPermissions] = useState(false);
```

Comment:
Nice! Do you think we could abstract this into something like a permission wrapper component?
I have no clue tbh I'm just wondering if we could apply this for other features